include/drivers/st25r3911b/spi.h

```diff
@@ -0,0 +1,163 @@
+ *
+ *  @details The register address is auto-incrementing.
+ *           Registers are read in the order ascending from
+ *           the starting address.
+ *
+ *  @param[in] start_reg The address of the register from which the
+ *                       reading will start. See @ref st25r3911b_reg
+ *                       for the register addresses.
+ *  @param[out] val Read registers values.
+ *  @param[in] len Number of registers to read.
+ *
+ *
+ *  @retval 0 If the operation was successful.
+ *            Otherwise, a (negative) error code is returned.
+ */
+int st25r3911b_multiple_reg_read(u8_t start_reg, u8_t *val, size_t len);
+
+/** @brief Read a single NFC Reader register.
+ *
+ *  @param[in] reg Register address (see @ref st25r3911b_reg).
+ *  @param[out] val Read register value.
+ *
+ *  @retval 0 If the operation was successful.
+ *            Otherwise, a (negative) error code is returned.
+ */
+static inline int st25r3911b_reg_read(u8_t reg, u8_t *val)
+{
+	return st25r3911b_multiple_reg_read(reg, val, 1);
+}
+
+/** @brief Execute an NFC Reader direct command.
+ *
+ *  @details Direct commands are used to automate certain activities.
+ *
+ *  @param[in] cmd Direct command (see @ref st25r3911b_direct_command).
+ *
+ *  @retval 0 If the operation was successful.
+ *            Otherwise, a (negative) error code is returned.
+ */
+int st25r3911b_cmd_execute(u8_t cmd);
+
+/** @brief Modify a single NFC Reader register.
+ *
+ *  @details The first register value is read,
+ *           then the bits selected by the clear mask are cleared.
+ *           Finally, the bits selected by the set mask are set.
+ *
+ *  @param[in] reg Register address (see @ref st25r3911b_reg).
+ *  @param[in] clr_mask Bitmask that defines which register
+ *                      bits to clear.
+ *  @param[in] set_mask Bitmask that defines which register
+ *                      bits to set.
+ *
+ *  @retval 0 If the operation was successful.
+ *            Otherwise, a (negative) error code is returned.
+ */
+int st25r3911b_reg_modify(u8_t reg, u8_t clr_mask, u8_t set_mask);
+
+/** @brief Read NFC Reader FIFO data.
+ *
+ *  @details Data from FIFO is read by SPI. The data length to
+ *           read can be in the range of 1 to 96 bytes.
+ *
+ *  @param[out] data Read FIFO data.
+ *  @param[in] length Length of data to read.
+ *
+ *  @retval 0 If the operation was successful.
+ *            Otherwise, a (negative) error code is returned.
+ */
+int st25r3911b_fifo_read(u8_t *data, size_t length);
+
+/** @brief Write NFC Reader FIFO data.
+ *
+ *  @details Data from FIFO is written by SPI.The  data length
+ *           to write can be in the range of 1 to 96 bytes.
+ *
+ *  @param[in] data  FIFO data to write.
+ *  @param[in] length Length of data to write.
+ *
+ *  @retval 0 If the operation was successful.
+ *            Otherwise, a (negative) error code is returned.
+ */
+int st25r3911b_fifo_write(u8_t *data, size_t length);
```

Comment:
`const u8_t *data`

Reply:
Spi buffer discard `const` so I think this is not needed here.